clang/lib/Sema/SemaOverload.cpp

```diff
@@ -7807,6 +7833,7 @@ void Sema::AddTemplateOverloadCandidate(
     Candidate.IgnoreObjectArgument =
         isa<CXXMethodDecl>(Candidate.Function) &&
         !isa<CXXConstructorDecl>(Candidate.Function);
+    Candidate.TookAddressOfOverload = false;
```

Comment:
Feel like Candidate should have a deleted default ctor/constructor pair too :D  For exactly this reason.